### DISCUSSION TITLE
TreeGridView Binding doesn't update with TextBoxCell

### DISCUSSION THREAD

**Discussion by WileyNg** [picoe/Eto#2284] [General]
https://user-images.githubusercontent.com/5993201/185616682-9d8e8441-a276-484e-ad4c-5536481415c6.mp4

I do not understand why the Binding doesn't update the TreeGridItem there with its Name changed.
Is it because there are too many levels?

My TreeGridItem.Tag holds a class called Globals.ControlObjectsProperty, under it i have a Name property.

Is it the correct way of doing it otherwise?

**Comment by cwensley** [comment]
It is because you're using [Convert](http://pages.picoe.ca/docs/api/html/M_Eto_Forms_IndirectBinding_1_Convert__1_1.htm), which has a `toValue` parameter (which you have specified) _and_ a `fromValue` parameter, which you have not.  So, it only knows how to convert the tag to the value required by the control, but not the other way around when it is entered.

Another way that might work a bit better is to use the [Child](http://pages.picoe.ca/docs/api/html/M_Eto_Forms_IndirectBinding_1_Child__1_1.htm) binding method, something like:

```c#
Binding.Property((TreeGridItem item) => item.Tag).Cast<Globals.ControlObjectsProperty>().Child(c => c.Name)
```

**Reply by WileyNg** [reply]
![image](https://user-images.githubusercontent.com/5993201/185714101-fd08906e-0cb7-4c69-a8e6-fb8b4c9c1c46.png)

Thanks so much for answer!
I tried this code, it gave me empty Names though. I am pretty sure all the items have names though.

                    Binding = Binding.Property((TreeGridItem item) => item.Tag)
                                     .Cast<Globals.ControlObjectsProperty>().Child(c=>c.Name),

Also, i do not know how I can convert string to Globals.ControlObjectsProperty using lambda here....

**Reply by WileyNg** [reply]
Hi @cwensley 
```C#
                Binding = Binding.Property((TreeGridItem item) => item.Tag)
                                 .Cast<Globals.ControlObjectsProperty>().Child(c=>c.Name),
```
This actually works!
The class has fields instead of exposed properties with get set.
This was a mistake on my part.
